config/m4/compiler.m4

```diff
@@ -364,23 +365,29 @@ AS_IF([test "x$with_avx" != xyes],
       ])
 
 
-DETECT_UARCH()
-
+# Only detect uarch if neither --with-mcpu nor --with-march is set
+AS_IF([test -z "$with_mcpu" -a -z "$with_march"], [
+    DETECT_UARCH()
+], [
+    # Always define these conditionals to avoid errors on non-ARM platforms
+    AM_CONDITIONAL([HAVE_AARCH64_THUNDERX2], [false])
+    AM_CONDITIONAL([HAVE_AARCH64_THUNDERX1], [false])
+    AM_CONDITIONAL([HAVE_AARCH64_HI1620], [false])
+])
 
 #
 # CPU tuning
 #
-AS_IF([test "x$ax_cpu" != "x"],
-      [COMPILER_CPU_OPTIMIZATION([mcpu], [CPU Model], [-mcpu=$ax_cpu],
+AS_IF([test "x$with_mcpu" != "x"],
+      [COMPILER_CPU_OPTIMIZATION([mcpu], [CPU Model], [-mcpu=$with_mcpu],
```

Comment:
1. seems the argument is added only if cpu is not auto-detected , with this pr we'd need to always add the argument
2. can you pls add build test for the new arg? to see it really takes affect